crates/bevy_ecs/src/system/commands.rs

```diff
@@ -24,7 +33,22 @@ impl CommandQueue {
     pub fn apply(&mut self, world: &mut World) {
         world.flush();
         for command in self.commands.drain(..) {
+            // TODO: replace feature by proper error handling from commands
+            // https://github.com/bevyengine/bevy/issues/2004
+            #[cfg(not(feature = "command_panic_origin"))]
             command.write(world);
+            #[cfg(feature = "command_panic_origin")]
+            {
+                let may_panic = panic::catch_unwind(AssertUnwindSafe(|| {
+                    command.write(world);
+                }));
+                if may_panic.is_err() {
+                    if let Some(system_name) = &self.system_name {
+                        error!("panic while applying a command from {}", system_name);
+                    }
+                    panic!("panic applying a command");
+                }
```

Comment:
```suggestion
                if let Err(err) = may_panic {
                    if let Some(system_name) = &self.system_name {
                        error!("panic while applying a command from {}", system_name);
                    }
                    std::panic::resume_unwind(err);
                }
```
prevents duplicate panic messages and backtraces.